### PR TITLE
RetryUtils: Ability to have some number of quiet retries.

### PR DESCRIPTION
**Pull request by gianm**
Inspired by the request in https://github.com/druid-io/druid/pull/2220#issuecomment-185864278.

### CODE REVIEW

File: src/main/java/com/metamx/common/RetryUtils.java [hunk 1]
```diff
@@ -28,7 +28,12 @@
 {
   public static final Logger log = new Logger(RetryUtils.class);
 
-  public static <T> T retry(final Callable<T> f, Predicate<Throwable> shouldRetry, final int maxTries) throws Exception
+  public static <T> T retry(
```
Comment (drcrallen) [inline src/main/java/com/metamx/common/RetryUtils.java]: Can you add javadocs?

Comment (gianm) [inline src/main/java/com/metamx/common/RetryUtils.java]: @drcrallen added docs

**Comment by drcrallen** [comment]
:+1:

**Comment by nishantmonu51** [comment]
:+1:

**Comment by gianm** [comment]
@drcrallen @nishantmonu51 could one of you do a release of java-util please? would like to depend on it for https://github.com/druid-io/druid/pull/2220#issuecomment-185864278.

**Comment by drcrallen** [comment]
@gianm released 0.27.9 (I borked the release process on 0.27.8... stupid javadoc in java8)

**Comment by gianm** [comment]
thanks @drcrallen!

**Pull request merged**